gnomad/assessment/sanity_checks.py

```diff
@@ -104,78 +118,86 @@ def make_filters_sanity_check_expr(
-        prefix = f"{prefix}_"
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
 
-    label_combos = make_label_combos(label_groups)
-    combo_AC = [ht.info[f"{prefix}AC_{x}"] for x in label_combos]
-    combo_AN = [ht.info[f"{prefix}AN_{x}"] for x in label_combos]
-    combo_nhomalt = [ht.info[f"{prefix}nhomalt_{x}"] for x in label_combos]
+    # Check if subset is an empty string which is added to check entire callset but does not need the added delimiter
+    if subset:
+        subset += delimiter
 
+    label_combos = make_label_combos(label_groups, label_delimiter=delimiter)
+    # Grab the adj group for checks as we do not retain the raw metric counts for all sample groups so we cannot check the sample sums
     group = label_groups.pop("group")[0]
-    alt_groups = "_".join(
+    # sum_group is the group that when their metrics are added together they should equal the metric total `group` e.g., if group = callset1, one sum_group could be callset1-pop
+    sum_group = delimiter.join(
         sorted(label_groups.keys(), key=lambda x: sort_order.index(x))
     )
+    info_fields = t.info.keys()
 
-    annot_dict = {
-        f"sum_AC_{group}_{alt_groups}": hl.sum(combo_AC),
-        f"sum_AN_{group}_{alt_groups}": hl.sum(combo_AN),
-        f"sum_nhomalt_{group}_{alt_groups}": hl.sum(combo_nhomalt),
-    }
+    # Loop through metrics to build dictionary of fields that make up 'group' and then stores their sum
+    annot_dict = {}
+    for metric in ["AC", "AN", "nhomalt"]:
+        sum_group_exprs = []
+        for label in label_combos:
+            if metric_first_label:
+                field = f"{metric}{delimiter}{subset}{label}"
+            else:
+                field = f"{subset}{metric}{delimiter}{label}"
 
-    ht = ht.annotate(**annot_dict)
+            if field in info_fields:
+                sum_group_exprs.append(t.info[field])
+            else:
+                logger.info("%s is not in table's info field", field)
 
-    for subfield in ["AC", "AN", "nhomalt"]:
-        if not subpop:
-            generic_field_check(
-                ht,
-                (
-                    ht.info[f"{prefix}{subfield}_{group}"]
-                    != ht[f"sum_{subfield}_{group}_{alt_groups}"]
-                ),
-                f"{prefix}{subfield}_{group} = sum({subfield}_{group}_{alt_groups})",
-                [
-                    f"info.{prefix}{subfield}_{group}",
-                    f"sum_{subfield}_{group}_{alt_groups}",
-                ],
-                verbose,
-            )
+        annot_dict[
+            f"sum{delimiter}{metric}{delimiter}{group}{delimiter}{sum_group}"
+        ] = hl.sum(sum_group_exprs)
+
+    field_check_expr = {}
+    for metric in ["AC", "AN", "nhomalt"]:
+        if metric_first_label:
+            check_field_left = f"{metric}{delimiter}{subset}{group}"
+            check_field_right = f"sum{delimiter}{metric}{delimiter}{subset}{group}{delimiter}{sum_group}"
         else:
```

Comment:
I just realized this could be further simplified
```suggestion
            check_field_left = f"{metric}{delimiter}{subset}{group}"
        else:
```